src/ServerPrivate.cc

```diff
@@ -207,67 +207,12 @@ void ServerPrivate::AddRecordPlugin(const ServerConfig &_config)
           //   SDF.)
           if (pluginName->GetAsString() == LoggingPlugin::RecordPluginName())
           {
-            std::string recordPath = _config.LogRecordPath();
-            std::string cmpPath = _config.LogRecordCompressPath();
-
-            // Set record path
-            if (!_config.LogRecordPath().empty())
```

Comment:
I think the `if (!_config.LogRecordPath().empty())` check is still needed, around the new lines 210-215, in case the string is empty.

Reply:
I removed it because it was making a test fail. But I made other changes to the test since then, so I can try adding it again.

Reply:
This is one of the tests that are broken if I check for an empty string:

https://github.com/ignitionrobotics/ign-gazebo/blob/d693bb636e09754ec16c782ff22d9dc6b93282ad/test/integration/log_system.cc#L331-L352

I'm confused about the expectations though, because recording does take place, just not to `<path>`.

Reply:
That `Recording will not take place` output looks right. Basically that's a case that would only happen if someone uses the C++ API to record and forgot to specify a path. In normal usage, the path is specified in `ign.cc`, so if they don't go through the command line, then the plugin doesn't know where to record to - and we decided not to initialize the ignLogDirectory inside the plugin.

Maybe some tests need to be changed since we remove the <path> in Dome, I will take a look. I have a library failure at runtime after I reinstalled using colcon and haven't been able to run the tests, ha... See if I can fix it.